pagopa/io-app
# chore: [IOBP-455] Removal of BonusVacanze references across the app

## Short description
removal the major references of BonusVacanze from the app

## List of changes proposed in this pull request
- removed navigation
- removed the specific map entries and feature flags
- extracted shared logic where needed

after this PR, the full removal of BonusVacanze should come much easier, although still with some hassle.

## How to test
run the app, and with the `dev-server` open:
- make sure it correctly runs
- make sure that CGN can onboard and work as intended
- make sure you can add cards as if you were in the `master` branch

overall, nothing should have changed from the `master` branch from an user's perspective.
